core/src/main/scala/org/apache/spark/sql/SnappyParser.scala

```diff
@@ -175,6 +176,19 @@ class SnappyParser(session: SnappySession)
     }
   }
 
+  private def updatePerPlanQueryHints(plan: LogicalPlan): LogicalPlan = {
+    if (queryHints.isEmpty) return plan
+    // check for join type hint
+    queryHints.remove(QueryHint.JoinType.toString) match {
+      case null => plan
+      case joinTypeHint => plan match {
+        case LogicalPlanWithHints(child, hints) =>
+          LogicalPlanWithHints(child, hints + (QueryHint.JoinType.toString -> joinTypeHint))
```

Comment:
How are we controlling the join type hint if multiple joins are there in a complex plan? Please explain a bit.